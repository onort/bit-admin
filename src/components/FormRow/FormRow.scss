@import "../../styles/main.scss";

// negative margin needed to replace form element margins collapsing
.row {
  display: flex;
  margin: (-$spacer-sm) 0;
}
