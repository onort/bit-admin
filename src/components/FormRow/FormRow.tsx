import React from "react"
import cx from "classnames"

import styles from "./FormRow.scss"

interface Props {
  children: React.ReactNode
  className?: string
}

const FormRow: React.SFC<Props> = props => {
  const className = cx(styles.row, props.className)
  return <div className={className}>{props.children}</div>
}

export default FormRow
