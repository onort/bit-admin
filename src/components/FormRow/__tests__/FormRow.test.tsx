import React from "react"
import { shallow } from "enzyme"

import FormRow from "../"

describe("<FormRow />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <FormRow>
        <span />
      </FormRow>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should be able to have custom class", () => {
    const testClassName = "test"
    const wrapper = shallow(
      <FormRow className={testClassName}>
        <span />
      </FormRow>
    )
    expect(wrapper.find("div").hasClass(testClassName)).toBe(true)
  })
})
