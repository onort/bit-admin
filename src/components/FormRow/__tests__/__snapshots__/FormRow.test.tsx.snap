// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormRow /> should match snapshot 1`] = `
<div
  className="row"
>
  <span />
</div>
`;

exports[`<FormRow /> should match snapshot 2`] = `"<div class=\\"row\\"><span></span></div>"`;
